python/parameters.rst
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| Parameter Name         | Type                                | Default                       | Documentation                                                                                          |
+========================+=====================================+===============================+========================================================================================================+
| h_int                  | Operator                            | --                            | Interacting part of the atomic Hamiltonian                                                             |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| n_cycles               | int                                 | --                            | Number of QMC cycles                                                                                   |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| partition_method       | str                                 | "autopartition"               | Partition method                                                                                       |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| quantum_numbers        | list(Operator)                      | []                            | Quantum numbers                                                                                        |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| length_cycle           | int                                 | 50                            | Length of a single QMC cycle                                                                           |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| n_warmup_cycles        | int                                 | 5000                          | Number of cycles for thermalization                                                                    |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| random_seed            | int                                 | 34788 + 928374 * MPI.rank     | Seed for random number generator                                                                       |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| random_name            | str                                 | ""                            | Name of random number generator                                                                        |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| max_time               | int                                 | -1 = infinite                 | Maximum runtime in seconds, use -1 to set infinite                                                     |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| verbosity              | int                                 | 3 on MPI rank 0, 0 otherwise. | Verbosity level                                                                                        |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| move_shift             | bool                                | true                          | Add shifting a move as a move?                                                                         |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| move_double            | bool                                | false                         | Add double insertions as a move?                                                                       |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| use_trace_estimator    | bool                                | false                         | Calculate the full trace or use an estimate?                                                           |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| measure_g_tau          | bool                                | true                          | Measure G(tau)?                                                                                        |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| measure_g_l            | bool                                | false                         | Measure G_l (Legendre)?                                                                                |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| measure_pert_order     | bool                                | false                         | Measure perturbation order?                                                                            |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| measure_density_matrix | bool                                | false                         | Measure the contribution of each atomic state to the trace?                                            |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| use_norm_as_weight     | bool                                | false                         | Use the norm of the density matrix in the weight if true, otherwise use Trace                          |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| performance_analysis   | bool                                | false                         | Analyse performance of trace computation with histograms (developers only)?                            |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| proposal_prob          | dict(str:float)                     | {}                            | Operator insertion/removal probabilities for different blocks                                          |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| move_global            | dict(str : dict(indices : indices)) | {}                            | List of global moves (with their names). Each move is specified with an index substitution dictionary  |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| move_global_prob       | double                              | 0.05                          | Overall probability of the global moves                                                                |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+
| imag_threshold         | double                              | 1.e-15                        | Threshold below which imaginary components of Delta and h_loc are set to zero                          |
+------------------------+-------------------------------------+-------------------------------+--------------------------------------------------------------------------------------------------------+

benchmark/5+5/params.py
#!/bin/env pytriqs

import pytriqs.utility.mpi as mpi

results_file_name = "5_plus_5.h5"
matrix_results_file_name = "5_plus_5.matrix.h5"
ref_file_name = "5_plus_5.ref.dat"

# Block structure of GF
L = 2 # d-orbital
cubic_names = ("xy","yz","z^2","xz","x^2-y^2")
mkind = lambda sn, cn: (sn+'_'+cn, 0)

# Input parameters
beta = 40.
mu = 26

use_interaction = True
U = 4.0
J = 0.7
F0 = U
# Sasha uses coefficient 0.63 instead of 0.625
F2 = J*(14.0/(1.0 + 0.63))
F4 = F2*0.63

# Dump the local Hamiltonian to a text file (set to None to disable dumping)
H_dump = "H.txt"
# Dump Delta parameters to a text file (set to None to disable dumping)
Delta_dump = "Delta_params.txt"

# Hybridization function parameters
# Delta(\tau) is diagonal in the basis of cubic harmonics
# Each component of Delta(\tau) is represented as a list of single-particle
# terms parametrized by pairs (V_k,\epsilon_k).
delta_params={"xy" : {'V':0.2,'e':-0.2},
              "yz" : {'V':0.2,'e':-0.15},
              "z^2" : {'V':0.2,'e':-0.1},
              "xz" : {'V':0.2,'e':0.05},
              "x^2-y^2" : {'V':0.2,'e':0.4}}

atomic_levels = {('up_xy',0) :        -0.2,
                 ('dn_xy',0) :        -0.2,
                 ('up_yz',0) :        -0.15,
                 ('dn_yz',0) :        -0.15,
                 ('up_z^2',0) :       -0.1,
                 ('dn_z^2',0) :       -0.1,
                 ('up_xz',0) :        0.05,
                 ('dn_xz',0) :        0.05,
                 ('up_x^2-y^2',0) :   0.4,
                 ('dn_x^2-y^2',0) :   0.4}

n_iw = 1025
n_tau = 10001

p = {}
p["max_time"] = -1
p["random_name"] = ""
p["random_seed"] = 123 * mpi.rank + 567
p["length_cycle"] = 50
p["n_warmup_cycles"] = 1000
p["n_cycles"] = 30000
p["partition_method"] = "autopartition"
p["measure_g_tau"] = True
p["move_shift"] = True
p["measure_pert_order"] = False
p["performance_analysis"] = False
p["use_trace_estimator"] = False
